### ISSUE TITLE
WriteDeviceResource allows any datatype to be written to a String type field

### ISSUE THREAD

**Issue by ajcasagrande** [edgexfoundry/device-sdk-go#1198]
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

device-sdk-go
device-onvif-camera

### Is this a regression?

Not sure

## 🔥 Exception or Error
Send a json object to a string value type:
![image](https://user-images.githubusercontent.com/48735240/180298690-1b5b5c11-1f44-4734-aae1-92beed9b7116.png)

See the value converted into a string containing the value type of `map`:
![image](https://user-images.githubusercontent.com/48735240/180298779-b45218f8-ec6c-43f8-8f91-a53641233b8d.png)

See where the problem is coming from. `value` is of `map` type, but `v` was created by `fmt.Sprint(value)`.
![image](https://user-images.githubusercontent.com/48735240/180298631-4869d7a5-30e5-431d-96cd-6572766cdeff.png)

thus causing validation to pass:
![image](https://user-images.githubusercontent.com/48735240/180299441-fdaa2c16-eb62-439f-a27a-e71cc8f21502.png)

<pre><code>

</code></pre>

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version [**REQUIRED**]:**

**Anything else relevant?**

**Comment by cloudxxx8** [comment]
In this example, it looks like the resource value type is defined in `String` from the Device Profile, so the service treats it as a pure String field.  If it's not string, users can define it as an object value type.
What is your expected behavior?
This can be an enhancement, not a but.  Also, this could be validated in the Core Command service to make sure Device Service always receives right data type.

**Comment by ajcasagrande** [comment]
@cloudxxx8 correct, the resource value type is defined as `String` in the device profile. The expected behavior is to return a 400 error back to the user, as they passed in a json object.
